src/sentry/grouping/strategies/newstyle.py

```diff
@@ -436,13 +436,21 @@ def _single_stacktrace_variant(
     values = []
     prev_frame = None
     frames_for_filtering = []
+    found_in_app_frame = False
+
     for frame in frames:
         with context:
             context["is_recursion"] = is_recursive_frames(frame, prev_frame)
             frame_component = context.get_single_grouping_component(frame, event=event, **meta)
 
-        if variant == "app" and not frame.in_app:
-            frame_component.update(contributes=False, hint="non app frame")
+        if variant == "app":
+            if frame.in_app:
+                found_in_app_frame = True
+            else:
+                # We have to do this here (rather than it being done in the rust enhancer) because
+                # the rust enhancer doesn't know about system vs app variants
+                frame_component.update(contributes=False, hint="non app frame")
```

Comment:
Should this be something like this?
```
if frame.in_app:
  found_in_app_frame = True

# We have to do this here (rather than it being done in the rust enhancer) because
# the rust enhancer doesn't know about system vs app variants
frame_component.update(
  contributes=found_in_app_frame,
  hint="foo" if found_in_app_frame else "non app frame"
)
```

Reply:
You'd need to replace `"foo"` in the last line with `None`, but otherwise I do believe that'd have the same effect as what I have. That said, you'd end up doing work when you don't need to:

- If we're dealing with the system variant, we don't care about in-app frames. (I'm not sure if you were suggesting keeping the `if variant == "app"` check or not.)

- `contributes` defaults to `True` and `hint` defaults to `None`, so if `frame.in_app` is `True`, the values are already what we want them to be. It's only out-of-app frames which need adjusting.

It's obviously not a huge performance hit just to set those two values, but we process a whole lot of frames, so in a tiny way it might add up. More importantly, though, I think doing it when we don't need to obscures the logic of what we're actually doing and why, so my preference is to leave it as is.